druid-shell/src/platform/mac/appkit.rs

```diff
@@ -73,3 +75,97 @@ impl NSView for id {
         msg_send![self, addTrackingArea: trackingArea]
     }
 }
+
+pub trait NSButton: Sized {
+    unsafe fn setKeyEquivalent(self, key: id);
+}
+
+impl NSButton for id {
+    unsafe fn setKeyEquivalent(self, key: id) {
+        msg_send![self, setKeyEquivalent: key]
+    }
+}
+
+pub trait NSControl: Sized {
+    unsafe fn setTag(self, tag: NSInteger);
+}
+
+impl NSControl for id {
+    unsafe fn setTag(self, tag: NSInteger) {
+        msg_send![self, setTag: tag]
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum NSAlertStyle {
+    Warning,
+    Informational,
+    Critical,
+}
+
+pub const NSModalResponseContinue: NSInteger = -1002;
```

Comment:
I often like to include a link to my source when I'm adding constants like this.

Reply:
I think that's a good idea in general, however I found these specific constants through a combination of testing+println and digging through random Objective-C/Swift apps on the internet. So no authoritative source to link.

I haven't found Apple docs that define constants like they are available from Microsoft for example. If there are any that exist, that would be great to know. A [few constants here and there are defined](https://developer.apple.com/documentation/appkit/nsalert/style/informational) on developer.apple.com but most aren't.

Reply:
so what I like to do here is grep in `/Library/Developer` and you'll often find the header file where something is declared, and these header files often include helpful documentation.